### ISSUE TITLE
Content Include `.feature` files in `.csproj`

### ISSUE THREAD

**Issue by konarx** [reqnroll/Reqnroll.Rider#1]
Related to [Discussion #57](https://github.com/orgs/reqnroll/discussions/57):

Hello!
After Reqnroll migration, the following snippet introduced in the `.csproj` file:
```xml
  <ItemGroup>
    <Content Include="Features\*.feature"/>
  </ItemGroup>
```
When using SpecFlow it wasn't there. And if I remove it, the IDE can not identify `.feature` files, only the `.feature.cs` files.

I am using Rider with the [latest build](https://github.com/reqnroll/Reqnroll.Rider/actions/runs/7951275864) of the Rider.Reqnroll plugin. 
Here is how it looks when having the above snippet in `.csproj` and how it should be looking even when remove it:
![image](https://github.com/reqnroll/Reqnroll/assets/23631145/b1bd04b1-37a2-4503-a581-9ca7f648beff)
and this is how it actually looks when removing it:
![image](https://github.com/reqnroll/Reqnroll/assets/23631145/d7d4f8b2-e153-4d76-9953-e7c95e8f3114)

**Comment by Socolin** [comment]
The plugin does not control what is shown here.

Rider is using the rules in the csproj (and the imported rules from from the Reqnroll nuget) to list this. If with specflow this is displayed and not with reqnroll then I would think the problem is in Reqnroll targets/properties rules.

@gasparnagy Do you know if there was there any change on that part on reqnroll and could you send me a link to where those files are in the reqnroll repository ?

**Comment by gasparnagy** [comment]
@Socolin there was no change in the target/property structure, but the properties have been renamed. 

I see such strange issues in VS, when the IDE is opened before the appropriate Reqnroll NuGet packages have been restored. 

@konarx could you please try to make a "dotnet restore" and open Rider only after that?

**Comment by Socolin** [comment]
So on a project still using specflow I don't have any problem, so I think something either was not rename correctly in the plugin or something changed in the csproj. It'll need more investigation

**Comment by Socolin** [comment]
So I took a look more deeply and the problem is only with Reqnroll, when using Specflow Nuget it works, even without any plugin so there is a problem somewhere in the .props, but this is weird because when I add the following snippet to the csproj rider detect them in double, so my guess is that rider find the `.props` correctly

```
    <ItemGroup>
        <ReqnrollFeatureFiles Include="**/*.feature" />
    </ItemGroup>
```

![image](https://github.com/reqnroll/Reqnroll.Rider/assets/539620/391d0a86-e057-4be0-b4ff-4797bfc00e84)

And without the `.feature` are only visible with the `Show All Files` options

![image](https://github.com/reqnroll/Reqnroll.Rider/assets/539620/5d9ae536-b5f6-4a33-a52c-18ebf231f48a)

The files can be found there

![image](https://github.com/reqnroll/Reqnroll.Rider/assets/539620/3e764955-b87e-4a8c-8a2f-80061b1d8dce)

----

I check the difference between both files (from SpecFlow and Reqnroll) and I was not able to find any meaningful difference, the only diff are on the `_Reqnroll_TaskFolder` with an updated version. (On this topic @gasparnagy I think this could be updated to `net6.0`, the folder seems to be available and netcoreapp3.1 is eol)

Also while I'm this file I think those line should be added (I did not have time to test this yet)

```
    <!-- FastUpToDateCheck for R# -->
    <!-- https://www.jetbrains.com/help/resharper/Building_Solution.html#supported-build-items -->
    <AvailableItemName Include="ReqnrollFeatureFiles"/>
```

**Comment by gasparnagy** [comment]
@Socolin Yes, fixing the net3.1 stuff is in progress. (SpecFlow has the same, so I don't think it has relevance.)

Maybe we should ask help from the Rider team?

**Comment by Socolin** [comment]
> @Socolin Yes, fixing the net3.1 stuff is in progress. (SpecFlow has the same, so I don't think it has relevance.)
> 
> Maybe we should ask help from the Rider team?

I did, and they told me we should build 2 projects (1 with specflow/reqnroll) with binlogs and try to find the differences.

**Comment by JohnYoungers** [comment]
I don't think it's an issue with the plugin; if I have a project with the following dependencies and the Rider plugin enabled:

```
<PackageReference Include="Reqnroll" Version="1.0.1" />
```

I can see the `*.feature` files fine (as well as the `*.feature.cs` files).  The moment I add this:

```
<PackageReference Include="Reqnroll.MsTest" Version="1.0.1"/>
```

The `*.feature` files vanish.  This 4 year old issue pointed me in the direction of it potentially being package related: https://github.com/SpecFlowOSS/SpecFlow/issues/1454.

The noted resolution from that issue of adding them via an `<ItemGroup>` node works, but I'd have to assume however this was resolved in specflow, could be resolved here as well

**Comment by teast** [comment]
I'm also experience this with Rider 2024.1.2 and Reqnroll plugin 2024.1.3.

One thing that I can add to the discussion is that with Reqnroll plugin + latest SpecFlow.NUnit (3.9.74 as of writing) Rider will present the feature file with the feature.cs file under it, that is as expected without adding anything to the csproj file.

I did try and go back in SpecFlow.NUnit versions and it was first at 3.3.57 that the *.feature.cs files started to be presented under their corresponding *.feature file. I'm not sure if this could be of any help in troubleshooting this..


Regards

**Comment by Drew99Kt** [comment]
<ItemGroup>
        <random Include="**/*.feature" />
    </ItemGroup>
    
    oddly enough this gets the feature files and behind files working on rider. 
    
specs
- 2024.2 EAP9 build 242.20224.162
- windows 10
- reqnroll plugin 2024.2.0

**Comment by Socolin** [comment]
Was this problem fixed ? I'm not able to reproduce

**Comment by savornicesei** [comment]
Does not seem to reproduce anymore on Rider 2025.3.3 and Renqroll plugin v2025.3.1. I think it can be closed and remove the reference to this ticket from documentation.

<img width="211" height="149" alt="Image" src="https://github.com/user-attachments/assets/ad5cd866-e91e-4928-99f5-e0c7aa46abb7" />
